simpleast-core/src/main/java/com/discord/simpleast/code/CodeRules.kt

```diff
@@ -208,6 +215,9 @@ object CodeRules {
 
         "xml" to xmlRules,
         "http" to xmlRules,
+
+        "js" to javascriptRules,
```

Comment:
Could these rules also apply to 
typescript? I would imagine most of the syntax overlaps?

Reply:
It should cover most of the syntax TypeScript provides but TS has type definitions which this PR does not include, as also definitions like `interface`, TS also has different and/or new keywords and built-ins and JavaScript itself does not include such as `public`, `private`, `protected` etc etc. Might make a different PR for TS i guess

Reply:
My thoughts were mainly that we might be able to have partial TSX support reusing some of the rules/keywords here. However, as you mentioned that's probably a different PR.